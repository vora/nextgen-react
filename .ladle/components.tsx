import type { GlobalProvider } from "@ladle/react";
import React from "react";
import { LadleErrorBoundary } from "./LadleErrorBoundary";
import "../src/styles.css";

export const Provider: GlobalProvider = ({ children }) => (
  <React.StrictMode>
    <LadleErrorBoundary>{children}</LadleErrorBoundary>
  </React.StrictMode>
);
