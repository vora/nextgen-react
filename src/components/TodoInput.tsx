import { useState } from "react";

const PlusIcon = () => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    className="h-6 w-6"
    fill="none"
    viewBox="0 0 24 24"
    stroke="currentColor"
    strokeWidth={2}
  >
    <path strokeLinecap="round" strokeLinejoin="round" d="M12 4v16m8-8H4" />
  </svg>
);

interface Props {
  onAdd?: (todo: string) => void;
}

export const TodoInput: React.FC<Props> = ({ onAdd }) => {
  const [value, setValue] = useState<string>("");

  const handleAdd = () => onAdd && onAdd(value);

  return (
    <div className="flex flex-col">
      <div className="flex flex-row">
        <input
          value={value}
          onChange={(e) => setValue(e.target.value)}
          placeholder="Add a TODO"
          className="p-4 rounded my-1 text-lg w-full"
        />
        <button
          onClick={handleAdd}
          className="border p-4 m-2 rounded shadow bg-white hover:bg-gray-200 active:bg-gray-400 flex flex-row items-center"
        >
          <PlusIcon />
        </button>
      </div>
    </div>
  );
};
