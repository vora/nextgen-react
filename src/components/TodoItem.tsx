export const TodoItem = () => {
  return <p>TODO Item</p>;
};
