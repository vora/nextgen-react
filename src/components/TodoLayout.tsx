import { ReactNode } from "react";

interface Props {
  children: ReactNode;
}

export const TodoLayout: React.FC<Props> = ({ children }) => {
  return (
    <div className="h-full w-full">
      <div className="border max-w-2xl rounded-lg shadow">
        <div className="bg-blue-500 p-4 text-center rounded-t-lg">
          <h1 className="text-white text-2xl font-bold">TODO List</h1>
        </div>
        <div className="p-4">{children}</div>
      </div>
    </div>
  );
};
