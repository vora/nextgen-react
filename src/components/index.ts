export * from "./TodoInput";
export * from "./TodoItem";
export * from "./TodoLayout";
