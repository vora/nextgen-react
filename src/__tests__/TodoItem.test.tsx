import renderer from "react-test-renderer";
import { TodoItem } from "~/components";

describe("TodoItem", () => {
  test("should match snapshot when checked", () => {
    const component = renderer.create(
      <TodoItem
        checked={true}
        onClick={() => null}
        onDelete={() => null}
        text="Checked Todo Item"
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("should match snapshot when unchecked", () => {
    const component = renderer.create(
      <TodoItem
        checked={false}
        onClick={() => null}
        onDelete={() => null}
        text="Unchecked Todo Item"
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
