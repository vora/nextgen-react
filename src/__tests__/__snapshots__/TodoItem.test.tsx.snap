// Vitest Snapshot v1

exports[`TodoItem > should match snapshot when checked 1`] = `
<div
  className="flex flex-row items-center justify-between"
>
  <p
    className="text-lg font-medium line-through text-gray-500"
  >
    Checked Todo Item
  </p>
  <div
    className="flex flex-row items-center"
  >
    <button
      className="text-xs text-red-700 border p-1 rounded"
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="ml-2"
      onClick={[Function]}
    >
      <svg
        className="fill-green-600"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`TodoItem > should match snapshot when unchecked 1`] = `
<div
  className="flex flex-row items-center justify-between"
>
  <p
    className="text-lg font-medium text-gray-800"
  >
    Unchecked Todo Item
  </p>
  <div
    className="flex flex-row items-center"
  >
    <button
      className="ml-2"
      onClick={[Function]}
    >
      <svg
        className="fill-blue-600"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </button>
  </div>
</div>
`;
